udev/50-picoprobe.rules

```diff
@@ -0,0 +1,8 @@
+# 2e8a:0004 Raspberry Pi picoprobe
+# https://github.com/raspberrypi/picoprobe
+SUBSYSTEM=="usb", ATTR{idVendor}=="2e8a", ATTR{idProduct}=="0004", MODE:="666"
```

Comment:
My preference really goes to using a group, rather than the satanic MODE.
But as long as the safer alternatives are suggested (line 7-8) I'm fine with this, as it's simpler for naïve users.
This is more a general comment, as all the .rules files here follow the same pattern.

Reply:
Yeah, that makes sense. I just followed the STLink rules that were created by ST. We could change all of the rules to use groups? Just need to add some instructions in comments on how to set up the group for users that don't know, and maybe the MODE based rules, commented out, as an alternative. This would be a separate PR, though.

Reply:
The good thing with the rules as they are now is that zero knowledge or extra actions are needed for the user, just drop them in and they work.
A group (or username) based rule needs the user to tailor it according their identity.

Probably, an username rule is easier - as it would not involve:
* Creating an ad-hoc group or selecting an existing one (but hey vary across distributions)
* Making sure the user belongs to the group

Only this step would be needed:
* put the username in the right place in the rule

I might try to write a template this WE, depending on available time - no guarantee.

Reply:
I'm sure that's why ST chose MODE based (and it made sense for me).

The ultimate would be to have all 3 variants: "just works", user based, group based. Separated into different folders under `./udev/`. That's probably overkill, though. 😁  If you find time, I will happily add your additions!